### ISSUE TITLE
TaskQueue.php Memory Limit

### ISSUE THREAD

**Issue by elvispdosreis** [guzzle/guzzle#2631]
**Guzzle version(s) affected**: 6.5.3
**PHP version**: 7.1.33
**cURL version**: 7.69.1

**Description**  

I have a recursive function that runs an http call several times until a row is empty, but I think that the guzzle is not emptying the memory

Fatal error: Allowed memory size of 58720256 bytes exhausted (tried to allocate 20480 bytes) in vendor/guzzlehttp/promises/src/TaskQueue.php on line 24.

function in a simplified way
```
public function updateProduct($page = 0, $limit = 100)
    {
        try {
            $response = $this->client->request('GET', "/products?page={$page}&per_page={$limit}");
            if ($response->getStatusCode() === 200) {
                $body = json_decode($response->getBody()->getContents());
                if ($page <= ceil($body->total / $limit)) {
                    $this->updateProduct($page + 1);
                }
            }
        } catch (Exception $e) {
            echo $e->getMessage() . PHP_EOL;;
        }
  }
```

**Comment by Nyholm** [comment]
Try with guzzle/promises:dev-master. I just fixed a memory leak there.

**Comment by Nyholm** [comment]
It is now released as guzzle/promises: 1.4
